### PR TITLE
Initial NATS publishing for load-balancer and load-balancer-ports

### PR DESCRIPTION
**Pull request by tylerauerbeck**
- adds initial NATS publishing for load-balancers and ports
- changes default nats-subject-prefix to `com.infratographer` (as we'll want to make sure that prefix.events and prefix.changes gets configured)
- `ent/schema` -> `ent/schema/schematype` to allow for schema hooks (and avoid the import cycle documented by ent)
- Flip pubsubx.Message -> pubsubx.ChangeMessage
- change subject format to `prefix.(changes|events).action.resource.location.locationid`

### CODE REVIEW

**Review by andy-v-h — DISMISSED** [review]

File: cmd/nats.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+		pubsub.WithLogger(logger),
+		pubsub.WithStreamName(viper.GetString("nats.stream-name")),
+		pubsub.WithSubjectPrefix(viper.GetString("nats.subject-prefix")),
+	)
+
+	_, err = client.AddStream()
+	if err != nil {
+		return &pubsub.Client{}, err
+	}
+
```
Comment (sfunkhouser) [inline cmd/nats.go:39]: nit: 
```suggestion
	if _, err = client.AddStream(); err != nil {
		return &pubsub.Client{}, err
	}

```

**Review by sfunkhouser — APPROVED** [review]

**Review by sfunkhouser — DISMISSED** [review]

File: cmd/nats.go (hunk 1 above)
Comment (tylerauerbeck) [reply cmd/nats.go]: ✅

File: internal/ent/schema/schematype/port.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+					if err := m.PubsubClient.PublishChange(ctx, types.OpToAction[m.Op()], types.TypeToSubject[gen.TypeLoadBalancer], location.String(), msg); err != nil {
+						err = errors.Wrap(err, "failed to publish event")
+						return val, err
+					}
+
+					return val, nil
+				})
+			},
+			// Limit the hook only for these operations.
+			ent.OpCreate|ent.OpUpdate|ent.OpDelete,
```
Comment (rizzza) [inline internal/ent/schema/schematype/port.go:181]: should add OpUpdateOne and OpDeleteOne too

Comment (tylerauerbeck) [reply internal/ent/schema/schematype/port.go]: ✅

File: internal/ent/schema/schematype/loadbalancer.go [hunk 3]
```diff
@@ -130,3 +138,69 @@ func (LoadBalancer) Annotations() []schema.Annotation {
...
+					if err := m.PubsubClient.PublishChange(ctx, types.OpToAction[m.Op()], types.TypeToSubject[gen.TypeLoadBalancer], location.String(), msg); err != nil {
+						err = errors.Wrap(err, "failed to publish event")
+						return val, err
+					}
+
+					return val, nil
+				})
+			},
+			// Limit the hook only for these operations.
+			ent.OpCreate|ent.OpUpdate|ent.OpDelete,
```
[the remark this answers is not in the document]
Comment (tylerauerbeck) [reply internal/ent/schema/schematype/loadbalancer.go]: ✅

**Review by rizzza — APPROVED** [review]

**Pull request closed without merging**